cupy/core/core.pyx

```diff
@@ -2890,6 +2901,80 @@ cpdef ndarray tensordot_core(
     return ret
 
 
+cpdef ndarray tensordot_core_v11(
+        Py_ssize_t transa, Py_ssize_t transb, Py_ssize_t m, Py_ssize_t n,
+        Py_ssize_t k, ndarray a, Py_ssize_t lda, ndarray b, Py_ssize_t ldb,
+        ndarray c, Py_ssize_t ldc):
+    cdef float one_f, zero_f
+    cdef double one_d, zero_d
+    cdef cuComplex one_F, zero_F
+    cdef cuDoubleComplex one_D, zero_D
+
+    if c.dtype.char in 'efF':
+        compute_type = cublas.CUBLAS_COMPUTE_32F
+    elif c.dtype.char in 'dD':
+        compute_type = cublas.CUBLAS_COMPUTE_64F
+
+    compute_capability = int(device.get_compute_capability())
+    algo = cublas.CUBLAS_GEMM_DEFAULT
+    if ((compute_capability >= 80) or
+            (compute_capability >= 70 and c.dtype == 'e')):
+        algo = cublas.CUBLAS_GEMM_DEFAULT_TENSOR_OP
+
+    a_cuda_dtype = dtype_to_cuda_dtype(a.dtype, is_half_allowed=True)
+    b_cuda_dtype = dtype_to_cuda_dtype(b.dtype, is_half_allowed=True)
+    c_cuda_dtype = dtype_to_cuda_dtype(c.dtype, is_half_allowed=True)
+    handle = device.get_cublas_handle()
+    if c.dtype.char in 'efd':
+        if compute_type == cublas.CUBLAS_COMPUTE_32F:
+            one_f = 1
+            zero_f = 0
+            cublas.gemmEx(
+                handle, <int>transa, <int>transb, <int>m, <int>n, <int>k,
+                <size_t>&one_f,
+                a.data.ptr, a_cuda_dtype, <int>lda,
+                b.data.ptr, b_cuda_dtype, <int>ldb,
+                <size_t>&zero_f, c.data.ptr, c_cuda_dtype, <int>ldc,
+                compute_type, algo)
+        elif compute_type == cublas.CUBLAS_COMPUTE_64F:
+            one_d = 1
+            zero_d = 0
+            cublas.gemmEx(
+                handle, <int>transa, <int>transb, <int>m, <int>n, <int>k,
+                <size_t>&one_d,
+                a.data.ptr, a_cuda_dtype, <int>lda,
+                b.data.ptr, b_cuda_dtype, <int>ldb,
+                <size_t>&zero_d, c.data.ptr, c_cuda_dtype, <int>ldc,
+                compute_type, algo)
```

Comment:
Looks like they can be combined?!
```suggestion
        if compute_type in (cublas.CUBLAS_COMPUTE_32F, cublas.CUBLAS_COMPUTE_64F):
            one = 1
            zero = 0
            cublas.gemmEx(
                handle, <int>transa, <int>transb, <int>m, <int>n, <int>k,
                <size_t>&one,
                a.data.ptr, a_cuda_dtype, <int>lda,
                b.data.ptr, b_cuda_dtype, <int>ldb                  
                <size_t>&zero, c.data.ptr, c_cuda_dtype, <int>ldc,
                compute_type, algo)
```

Reply:
It would be nice to be able to do so, but the dtypes of the parameter `alpha` and `beta` of cublasGemmEx (in this case, `one` and `zero`) have to be `float` pointer when compute type is `COMPUTE_32F` and `double` pointer when compute type is `COMPUTE_64F`.
https://docs.nvidia.com/cuda/cublas/index.html#cublas-GemmEx

Reply:
Ah, certainly, there is the way to do it. Thanks @asi1024 !